core/resources.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// Think of Resources as File Descriptors. They are integers that are allocated
+// by the privileged side of Deno to refer to various resources.  The simplest
+// example are standard file system files and stdio - but there will be other
+// resources added in the future that might not correspond to operating system
+// level File Descriptors. To avoid confusion we call them "resources" not "file
+// descriptors". This module implements a global resource table. Ops (AKA
+// handlers) look up resources by their integer id here.
+
+use downcast_rs::Downcast;
+use std;
+use std::any::Any;
+use std::collections::BTreeMap;
+use std::io::Error;
+use std::io::ErrorKind;
+
+/// Also referred to as rid.
```

Comment:
ResourceId is Deno's version of a file descriptor. ResourceId is also referred to as rid in the code base.